hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/timeline/versioning/v1/TimelineArchiverV1.java

```diff
@@ -442,9 +443,14 @@ private void writeToFile(Schema wrapperSchema, List<IndexedRecord> records) thro
       Map<HeaderMetadataType, String> header = new HashMap<>();
       header.put(HoodieLogBlock.HeaderMetadataType.SCHEMA, wrapperSchema.toString());
       final String keyField = table.getMetaClient().getTableConfig().getRecordKeyFieldProp();
-      List<HoodieRecord> indexRecords = records.stream().map(HoodieAvroIndexedRecord::new).collect(Collectors.toList());
-      HoodieAvroDataBlock block = new HoodieAvroDataBlock(indexRecords, header, keyField);
-      writer.appendBlock(block);
+      List<HoodieRecord> indexRecords = records.stream()
+          .filter(Objects::nonNull)
```

Comment:
how could this happen then, in 8 -> 7 downgrade, the `MetadataConversionUtils.createMetaWrapper` always return a `IndexedRecord` there?